src/java/com/zimbra/oauth/utilities/OAuthDataSource.java

```diff
@@ -138,20 +147,25 @@ public void syncDatasource(ZMailbox mailbox, OAuthInfo credentials) throws Servi
                 dsAttrs.put(Provisioning.A_zimbraDataSourceOAuthRefreshToken, refreshToken);
                 dsAttrs.put(Provisioning.A_zimbraDataSourceHost, host);
                 dsAttrs.put(Provisioning.A_zimbraDataSourceImportOnly, "FALSE");
-                if (importClassMap.containsKey(View.contact.name())) {
-                    // define the import class and polling interval
-                    ZimbraLog.extensions.debug("Setting datasource polling interval and import class.");
+                // define the import class and polling interval
+                ZimbraLog.extensions.debug("Setting datasource polling interval and import class.");
+                if (dsType == DataSourceType.oauth2contact && importClassMap.containsKey(View.contact.name())) {
```

Comment:
Use `importClassMap.containsKey(view.name)` and `importClassMap.get(view.name())` in the if statement and the put statement.

There is no need to check the dsType here since we will be mapping to it from the view type. No need for else if.